prereise/gather/demanddata/transportation_electrification/dwelling.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import math
+
+
+def get_segment(dwelling_start_time, dwelling_length):
+    """Get dwelling activity segment.
+
+    :param float dwelling_start_time: dwelling start time.
+    :param float dwelling_length: dwelling end time.
+    :return: (*int*) -- the amount of the rates (cost function) segments the dwelling activity possess.
+    """
+    total_dwell_period = dwelling_start_time + dwelling_length
+    return np.floor(total_dwell_period) - np.floor(dwelling_start_time) + 1
+
+
+def get_energy_limit(
+    power, segment, dwelling_start_time, dwelling_length, charging_efficiency
+):
+    """Determines the energy limit for the entire dwelling period. Takes into
+    consideration if charging does not start or end directly on the hour.
+
+    :param int power: charger power, EVSE kW.
+    :param int segment: the amount of the rates segments the dwelling activity possess.
+    :param float dwelling_start_time: dwelling start time.
+    :param float dwelling_length: dwelling end time.
```

Comment:
Based on the context, seems like this should dwelling length?

Reply:
Yup! That was an old description, it's been updated now.